packages/lesswrong/components/tagging/TagFlagAllPagesItem.tsx

```diff
@@ -0,0 +1,81 @@
+import React from "react"
+import { Tags } from "../../lib/collections/tags/collection";
+import { useMulti } from "../../lib/crud/withMulti";
+import { Components, registerComponent } from "../../lib/vulcan-lib";
+import classNames from 'classnames';
+import { useHover } from "../common/withHover";
+import { AnalyticsContext } from "../../lib/analyticsEvents";
+import Card from "@material-ui/core/Card";
+import { commentBodyStyles } from "../../themes/stylePiping";
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    ...theme.typography.commentStyle,
+    padding: 4,
+    margin: 4,
+    borderRadius: 5,
+    backgroundColor: 'rgba(0,0,0,0.1)',
+    display: 'inline-block'
+  },
+  black: {
+    color: 'white',
+    backgroundColor: 'rgba(0,0,0,0.8)'
+  },
+  white: {
+    backgroundColor: 'white',
+    border: '1px solid rgba(0,0,0,0.4)',
+    color: 'rgba(0,0,0,0.6)'
+  },
+  hoverCard: {
+    maxWidth: 350,
+    padding: theme.spacing.unit,
+    ...commentBodyStyles(theme)
+  }
+})
+
+const TagFlagAllPagesItem = ({ showNumber = true, style = "grey", classes }: {
```

Comment:
This... is just a 90% copy-paste of TagFlatItem? Why would we do this? This has to be merged with the existing TagFlagItem, this level of code-duplication is a giant pain to maintain.

Reply:
Yeah, I very much expected you to say that.

Reply:
This is why I marked it WIP :P